homeassistant/components/ista_ecotrend/coordinator.py

```diff
@@ -35,20 +35,27 @@ def __init__(self, hass: HomeAssistant, ista: PyEcotrendIsta) -> None:
     async def _async_update_data(self):
         """Fetch ista EcoTrend data."""
 
-        if not self.details:
-            self.details = await self.async_get_details()
-
         try:
+            await self.hass.async_add_executor_job(self.ista.login)
+
+            if not self.details:
+                self.details = await self.async_get_details()
+
             return await self.hass.async_add_executor_job(self.get_consumption_data)
+
         except ServerError as e:
             raise UpdateFailed(
                 "Unable to connect and retrieve data from ista EcoTrend, try again later"
             ) from e
         except (LoginError, KeycloakError) as e:
+            if TYPE_CHECKING:
+                assert self.config_entry
```

Comment:
Instead of type checking here you could overwrite the type directly on the class as this always will have a config entry

Reply:
```python
class IstaCoordinator(DataUpdateCoordinator[dict[str, Any]]):
    """Ista EcoTrend data update coordinator."""

    config_entry: ConfigEntry

    def __init__(self, hass: HomeAssistant, ista: PyEcotrendIsta) -> None:
```